provider/pkg/provider/deployment_settings.go

```diff
@@ -52,10 +66,43 @@ func (ds *PulumiServiceDeploymentSettingsInput) ToPropertyMap() resource.Propert
 				if ds.SourceContext.Git.GitAuth.SSHAuth != nil {
 					sshAuthPropertyMap := resource.PropertyMap{}
 					if ds.SourceContext.Git.GitAuth.SSHAuth.SSHPrivateKey.Value != "" {
-						sshAuthPropertyMap["sshPrivateKey"] = resource.NewPropertyValue(FixMe)
+						if mergeMode {
+							var plaintextValue *pulumiapi.SecretValue
+							var currentCipherValue *pulumiapi.SecretValue
+							if currentStateCipherSettings.SourceContext != nil &&
+								currentStateCipherSettings.SourceContext.Git != nil &&
+								currentStateCipherSettings.SourceContext.Git.GitAuth != nil &&
+								currentStateCipherSettings.SourceContext.Git.GitAuth.SSHAuth != nil {
+								plaintextValue = &plaintextInputSettings.SourceContext.Git.GitAuth.SSHAuth.SSHPrivateKey
```

Comment:
Is this guaranteed to exist? What happens if `plaintextInputSettings.SourceContext.Git.GitAuth.SSHAuth` is `nil`?

Reply:
Yes, if a value in plaintextInputSettings exists, it also exists in plaintextInputSettings, because they are inputs and outputs of the current state

Reply:
I'm guessing you meant "if a value in currentStateCipherSettings exists, it also exists in plaintextInputSettings"?

Reply:
Ah, yes, sorry!